pkg/config/signer.go

```diff
@@ -35,44 +34,44 @@ type ecdsaSignature struct {
 	R, S *big.Int
 }
 
-// NewSigner creates a new signer for configuration updates. The publicCert
-// and privateKey should be pem encoded.
-func NewSigner(publicCert []byte, privateCert []byte, mspID string) (*Signer, error) {
+// NewSigningIdentity creates a new signingIdentity for configuration updates.
```

Comment:
done

Reply:
+1 @wlahti The term `signer` and `signer's public/private` key are widely used as far as I know based on the famous crypto book that I have been reading -- `Applied Cryptography: Protocols, Algorithms and Source Code in C`. Hence, `Signer` looks more natural to me than `SigningIdentity` here. This is similar to having `Validator` or `Committer` as the struct name (which we have used).

Reply:
A signer signs things; this is more than a signer as it also includes the MSPID and the certificate that goes into the the signature header. The overloaded use of “signer” also adds confusion with respect to `crypto.Signer` from the standard library.

Within Fabric, across components, we refer to this collection of identity, cert, and key as a `SigningIdentity`. Hence the original request to change.

https://github.com/hyperledger/fabric/blob/43ff898e6c3bfe05876a600a878603c02d5094c2/msp/msp.go#L78

I believe Will’s comment was that the godoc should use the term “signing identity” instead of the type; if the type is used, it should at least be done in consistent case. He can clarify if I’m wrong.

Reply:
Correct, I was simply referring to the godoc in my previous comment. Things look good to me now.

Reply:
Got it. Thanks, @sykesm, for the detailed explanation.

Reply:
Even though **GetSigningIdentity** exists in the MSP interface, in my opinion it creates an un-needed coupling between signing and verification.

What if you have an idemix MSP for a specific organization **OrgA** but not for the rest of the organizations?  If a peer from **OrgB** now gets an API call to verify an idemix signature it needs to ask from the idemix MSP, which means an instance of the said MSP exists, however - the implementation of the idemix MSP on any org other than **OrgA** is undefined since only **OrgA** can issue idemix proofs.

If you ask me - the signing part should have never been placed inside the MSP interface, but it should have been in its own interface.